src/Bridge.ts

```diff
@@ -781,6 +781,12 @@ export class Bridge {
         if (event.state_key !== this.as.botUserId) {
             return this.as.botIntent.underlyingClient.kickUser(this.as.botUserId, roomId, "Bridge does not support DMing ghosts");
         }
+
+        // Don't accept invites from people who can't do anything
+        if (this.config.checkPermissionAny(event.sender, BridgePermissionLevel.login)) {
+            return this.as.botIntent.underlyingClient.kickUser(this.as.botUserId, roomId, "You do not have permission to inite this bot");
```

Comment:
FWIW, some clients (we all know which one) don't render the content of room leaves, so consider having the bot accept the join, post a goodbye message, and then leave the room.  (If this is done, there may have to be something to cleanly handle all of the bot's `"room.join"` callbacks.)

If it's not important to display a message, this can instead just use `leaveRoom`.

Also, may as well convert `botIntent.underlyingClient` to `botClient` for better readability.

Reply:
Actually, `botIntent.underlyingClient` appears all throughout the code, so maybe there can be a dedicated PR to change all of them at once.

Reply:
>don't render the content of room leaves, so consider having the bot accept the join, post a goodbye message, and then leave the room

I'd consider that a client that doesn't conform to the spec. Joins are **not** free (imagine being a small homeserver getting an invite to HQ), so the reason for a leave is to avoid that expensive operation.